src/App/VPSScan/EmbeddedBinary.hs

```diff
@@ -0,0 +1,73 @@
+{-# LANGUAGE TemplateHaskell #-}
+
+module App.VPSScan.EmbeddedBinary 
+  ( withUnpackedSherlockCli
+  , withUnpackedIPRClis
+  , IPRBinaryPaths(..)
+  ) where
+
+import Prelude hiding (writeFile)
+import Control.Effect.Exception
+import Control.Monad.IO.Class
+import Data.ByteString (writeFile, ByteString)
+import System.FilePath ((</>))
+import Path hiding ((</>))
+import Path.IO
+import Data.FileEmbed.Extra
+
+withUnpackedSherlockCli :: (Has (Lift IO) sig m, MonadIO m) => (FilePath -> m a) -> m a
+withUnpackedSherlockCli act =
+  bracket (liftIO getTempDir >>= \tmp -> createTempDir tmp "fossa-vpscli-vendor-sherlock")
+          (liftIO . removeDirRecur)
+          go
+  where
+    go tmpDir = do
+      let binaryPath = fromAbsDir tmpDir </> "sherlock-cli"
+      liftIO (writeFile binaryPath embeddedBinarySherlockCli)
+      liftIO (makeExecutable binaryPath)
+      act binaryPath
+
+data IPRBinaryPaths = IPRBinaryPaths
+  { ramjetBinaryPath :: FilePath
```

Comment:
[nit] [not blocking] I'd prefer these to be typed paths -- `Path Abs File`, probably. this helps reinforce that it's safe to `exec` this script from _any_ directory, because it's not a relative path to a script

I know I don't do it in the maven plugin module, but I don't know what past-me was thinking